core/trino-main/src/main/java/io/trino/execution/buffer/OutputBufferMemoryManager.java

```diff
@@ -54,18 +58,30 @@
     @GuardedBy("this")
     private ListenableFuture<Void> blockedOnMemory = NOT_BLOCKED;
 
+    private final Ticker ticker = Ticker.systemTicker();
+
     private final AtomicBoolean blockOnFull = new AtomicBoolean(true);
 
     private final Supplier<LocalMemoryContext> memoryContextSupplier;
     private final Executor notificationExecutor;
 
+    private final AtomicReference<TDigestHistogram> finalOutputBufferUtilization = new AtomicReference<>(new TDigestHistogram(new TDigest()));
```

Comment:
It would be great to have in-flight utilization stats rather than wait for query (task) to finish.